### ISSUE TITLE
Upgrade to use kubernetes api v1

### ISSUE THREAD

**Issue by tibcoplord** [m88i/nexus-operator#190]
**Describe the bug**
With recent kubectl we see warnings about using deprecated CustomResourceDefinition and ClusterRole beta apis.  We should upgrade to v1.

**To Reproduce**

```
$ kubectl apply -f nexus-operator.yaml 
...
namespace/nexus-operator-system created
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
...
clusterrole.rbac.authorization.k8s.io/nexus-operator-proxy-role created
Warning: rbac.authorization.k8s.io/v1beta1 ClusterRole is deprecated in v1.17+, unavailable in v1.22+; use rbac.authorization.k8s.io/v1 ClusterRole
```

**Expected behavior**
No warnings generated

**Environment**

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T18:49:28Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:41:49Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by ricardozanini** [comment]
Hi @tibcoplord!

Thanks for reporting! Yes, we are targeting this upgrade to 0.5.0 since we would like to still give support for old clusters like OpenShift 3.11. 0.4.0 is now cluster scoped, so it's a good feature to have for everyone out there.

Is this blicking you to use the operator?

Thanks again!

**Comment by tibcoplord** [comment]
> Is this blocking you to use the operator?

No .. I just spotted it so thought best to report.

**Comment by ricardozanini** [comment]
Sure! Many thanks for reporting it :)

**Comment by ricardozanini** [comment]
Fixed in #215
